dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/config/TrackerValidationConfig.java

```diff
@@ -50,60 +72,77 @@
 @Configuration( "trackerImportValidationConfig" )
 public class TrackerValidationConfig
 {
-    @Bean( "ruleEngineValidationHooks" )
-    public List<Class<? extends TrackerValidationHook>> getRuleEngineValidationHooks()
+    private final Map<Class<? extends TrackerValidationHook>, TrackerValidationHook> validationHooks;
+
+    public TrackerValidationConfig( Collection<TrackerValidationHook> checkers )
+    {
+        validationHooks = byClass( checkers );
+    }
+
+    @SuppressWarnings( "unchecked" )
+    private <T> Map<Class<? extends T>, T> byClass( Collection<T> items )
+    {
+        return items.stream()
+            .collect( Collectors.toMap(
+                e -> (Class<? extends T>) e.getClass(),
+                Functions.identity() ) );
+    }
+
+    @Bean
+    public List<TrackerValidationHook> ruleEngineValidationHooks()
     {
         return ImmutableList.of(
-            EnrollmentRuleValidationHook.class,
-            EventRuleValidationHook.class,
-            TrackedEntityAttributeValidationHook.class,
-            EnrollmentAttributeValidationHook.class,
-            EventDataValuesValidationHook.class );
+            getHookByClass( EnrollmentRuleValidationHook.class ),
+            getHookByClass( EventRuleValidationHook.class ),
+            getHookByClass( TrackedEntityAttributeValidationHook.class ),
+            getHookByClass( EnrollmentAttributeValidationHook.class ),
+            getHookByClass( EventDataValuesValidationHook.class ) );
     }
 
-    @Bean( "validationOrder" )
-    public List<Class<? extends TrackerValidationHook>> getValidationOrder()
+    @Bean
+    public List<TrackerValidationHook> validationHooks()
     {
         return ImmutableList.of(
-            PreCheckUidValidationHook.class,
-            PreCheckExistenceValidationHook.class,
-            PreCheckMandatoryFieldsValidationHook.class,
-            PreCheckMetaValidationHook.class,
-            PreCheckUpdatableFieldsValidationHook.class,
-            PreCheckDataRelationsValidationHook.class,
-            PreCheckSecurityOwnershipValidationHook.class,
+            getHookByClass( PreCheckUidValidationHook.class ),
```

Comment:
If we apply the same method to all values I think it makes sense to `map` instead.